### PR TITLE
first end point completed with error handling and tests

### PR DESCRIPTION
**Pull request by nickWoott**
First endpoint completed. 
Tested for correct response.
Tested erro handling. 
Also added 500 server error.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+describe.only("GET/api/topics", () => {
+  test("200, response with an object", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((res) => {
+        expect(typeof res.body).toBe("object");
+      });
+  });
```
Comment (xmenbteam) [inline __tests__/app.test.js:21]: We're not expecting an object, we're expecting an array. How do you test for an array?

**Review by xmenbteam — CHANGES_REQUESTED** [review]
Have a look at the comments, nearly there!

File: __tests__/app.test.js (hunk 1 above)
Comment (nickWoott) [reply __tests__/app.test.js:21]: changed it to toBeInstanceOf(Array)

File: topics.models.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+exports.selectTopics = () => {
+  return db
+    .query("SELECT * FROM topics")
+    .then((results) => {
+      console.log(results.rows, "<< model log");
+      return results.rows;
+    })
+    .catch((err) => {
+      console.log("error in the model");
+    });
```
[the remark this answers is not in the document]
Comment (nickWoott) [reply topics.models.js:13]: Whoops. This shouldn't be here. This was left over from when earlier. Deleted.

File: __tests__/app.test.js (hunk 1 above)
Comment (xmenbteam) [reply __tests__/app.test.js:21]: Nice!

**Pull request merged**